spacy_llm/cache.py

```diff
@@ -68,11 +68,11 @@ def __init__(
 
         self._init_cache_dir()
 
-    def initialize(self, vocab: Vocab, task: LLMTask) -> None:
+    def initialize(self, vocab: Vocab, task: ShardingLLMTask) -> None:
```

Comment:
The cache is still generic enough to also work with the normal task types, no? i.e. `Union[NonshardingLLMTask, ShardingLLMTask]`?